File: store_data_at_delete-setting-case-java/src/main/java/com/example/demo/memo/MemoHistoryResponse.java
package com.example.demo.memo;

import com.fasterxml.jackson.annotation.JsonFormat;
import org.springframework.data.history.RevisionMetadata;

import java.time.LocalDateTime;

public record MemoHistoryResponse(
        Long revId,
        String revType,

        Long id,
        String title,
        String content,

        @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss") LocalDateTime createdAt,
        @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss") LocalDateTime updatedAt
) {
    public static MemoHistoryResponse fromEntity(MemoEntity entity, RevisionMetadata<Long> metadata) {
        return new MemoHistoryResponse(
                metadata.getRequiredRevisionNumber(),
                metadata.getRevisionType().name(),
                entity.getId(),
                entity.getTitle(),
                entity.getContent(),
                entity.getCreatedAt(),
                entity.getUpdatedAt()
        );
    }
}

File: store_data_at_delete-setting-case/src/main/kotlin/com/example/demo/memo/MemoHistoryResponse.kt
package com.example.demo.memo

import com.fasterxml.jackson.annotation.JsonFormat
import org.springframework.data.history.RevisionMetadata
import java.time.LocalDateTime

data class MemoHistoryResponse(
    val revId: Long,
    val revType: String,

    val id: Long,
    val title: String,
    val content: String,

    @field:JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    val createdAt: LocalDateTime,

    @field:JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    val updatedAt: LocalDateTime,
)

fun MemoEntity.toHistoryResponse(metadata: RevisionMetadata<Long>): MemoHistoryResponse {
    return MemoHistoryResponse(
        revId = metadata.requiredRevisionNumber,
        revType = metadata.revisionType.name,
        id = id,
        title = title,
        content = content,
        createdAt = createdAt,
        updatedAt = updatedAt,
    )
}
